openedx/core/djangoapps/auth_exchange/views.py

```diff
@@ -53,16 +63,11 @@ def post(self, request, _backend):
         form = AccessTokenExchangeForm(request=request, oauth2_adapter=self.oauth2_adapter, data=request.POST)  # lint-amnesty, pylint: disable=no-member
         if not form.is_valid():
             error_response = self.error_response(form.errors)  # pylint: disable=no-member
-            if error_response.status_code == 403:
-                log.info('message=login_filed_1, status="%d", user="%s" ,agent="%s", error_response"%s"',
-                         error_response.status_code,
-                         request.user,
-                         request.META.get('HTTP_USER_AGENT', ''),
-                         error_response.data
-                         )
             return error_response
 
         user = form.cleaned_data["user"]
+        if request.user and not request.user.is_anonymous and request.user != user:
+            return self._get_invalid_request_response('session user and token user do not match')
```

Comment:
Why allow session auth at all on this endpoint?  Why restrict it to only consume token auth?

Reply:
> Why allow session auth at all on this endpoint? 
I think I plan on dropping this. I added it because I thought it was odd (and potentially might indicate a bug) if someone both presents a session cookie for one user and a token for a different user.  I still think this is true, but possibly not worth the added complexity and we could just ignore this case.

> Why restrict it to only consume token auth?
I'm not clear what this question means.